### DISCUSSION TITLE
How to change HTTP header before seleniumbase goes to website

### DISCUSSION THREAD

**Discussion by Vrackko** [seleniumbase/SeleniumBase#2656] [Q&A] [answered]
there is no option in seleniumbase to change http header settings. Is that even possiable or? I found a way with this code but seleniumbase doesnt have "add_headers." I need to change 'sec-ch-ua-platform: "Windows"' to 'sec-ch-ua-platform: "Android"'

`from seleniumbase import BaseCase

class MyTestClass(BaseCase):
    def test_custom_headers(self):
        custom_headers = {
            'User-Agent': 'Custom User Agent',
            'Custom-Header': 'Custom Value'
        }

        self.add_headers(custom_headers)
        self.open('http://example.com')`

**Comment by mdmintz** [comment, marked as answer]
See https://github.com/seleniumbase/SeleniumBase/issues/1906#issuecomment-1584516315
Also see https://github.com/seleniumbase/SeleniumBase/discussions/1179

**Reply by Vrackko** [reply]
Doesnt work for me. Example:
`    try:
        driver.execute_cdp_cmd(
        "Network.setExtraHTTPHeaders",
        {
            "headers": {
                "sec-ch-ua-platform": ""
            }
        }
        )
        print("Header changed")
    except Exception as e:
        print("Header change failed with error:", e)`

**Reply by Vrackko** [reply]
whatever value i put there it doesnt work

**Reply by mdmintz** [reply]
Use the `SB(wire=True)` / `--wire` mode for the `selenium-wire` integration:
https://github.com/wkeeling/selenium-wire#example-add-a-request-header

But if you have any issues specifically with `selenium-wire`, note that it's an external repo that I cannot change.
